src/modules/esl-event-listener/test/listener.resubscribtion.test.ts

```diff
@@ -0,0 +1,47 @@
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {ESLEventUtils} from '../core/api';
+import {listen} from '../../esl-utils/decorators/listen';
+
+describe('ESLEventListener re-subscribtion usecase with condition', () => {
+  class TestSubsEl extends ESLBaseElement {
+    public static override is = 'test-subs-el';
+    public mockFn = jest.fn();
+
+    get enabled() {
+      return this.hasAttribute('enabled');
+    }
+    set enabled(value) {
+      this.$$attr('enabled', value);
+      this.$$off(this.onClick);
+      this.$$on(this.onClick);
+    }
+
+    @listen({event: 'click', condition: (that: TestSubsEl) => that.enabled})
+    onClick() {
+      this.mockFn();
+    }
+  }
+
+  TestSubsEl.register();
+
+  const host = TestSubsEl.create();
+  test('ESLEventListener initial subscription is rejected by condition', () => {
```

Comment:
very-very cosmetic: pls, add a space before ‘test’, otherwise it's hard to see it